examples/portable/executor_runner/executor_runner.cpp

```diff
@@ -151,8 +164,20 @@ int main(int argc, char** argv) {
   // the method can mutate the memory-planned buffers, so the method should only
   // be used by a single thread at at time, but it can be reused.
   //
-
-  Result<Method> method = program->load_method(method_name, &memory_manager);
+  EventTracer* event_tracer_ptr = nullptr;
+#ifdef ET_EVENT_TRACER_ENABLED
```

Comment:
This function is already so long and complex, I'd like to factor out these ifdefs if possible.

You could create a class to encapsulate the event tracing, like

```
class TraceManager {
 public:
  TraceManager();
  EventTracer* get_event_tracer();
  Error write_etdump_to_file(const char* filename);
};
```

If tracing is enabled, the ctor could create the ETDump (a field), get_event_tracer can return a pointer to it, and write_to_file can open the file and write the contents. If tracing is disabled, the class is basically empty, returning a null tracer and just returning Error::NotSupported when asked to write.

Then main() can say
```
TraceManager tracer;
program->load_method(..., tracer->get_event_tracer());
...
if (tracer->get_event_tracer() != nullptr) {
  status = tracer->write_etdump_to_file(FLAGS_etdump_path.c_str());
  ET_CHECK_MSG(status == Error::Ok, ...);
}
```